diff -u -u -r a/SoObjects/SOGo/LDAPSource.h b/SoObjects/SOGo/LDAPSource.h
--- a/SoObjects/SOGo/LDAPSource.h	2017-03-24 14:46:32.000000000 +0100
+++ b/SoObjects/SOGo/LDAPSource.h	2017-03-31 03:26:12.197490190 +0200
@@ -75,6 +75,7 @@
   NSDictionary *modulesConstraints;
 
   NSMutableArray *searchAttributes;
+  NSArray *LDAPLookupFields;
   
   BOOL passwordPolicy;
   BOOL updateSambaNTLMPasswords;
@@ -110,6 +111,7 @@
     IMAPLoginField: (NSString *) newIMAPLoginField
     SieveHostField: (NSString *) newSieveHostField
         bindFields: (id) newBindFields
+  LDAPLookupFields: (NSArray *) newLDAPLookupFields
 	 kindField: (NSString *) newKindField
 andMultipleBookingsField: (NSString *) newMultipleBookingsField;
 
diff -u -u -r a/SoObjects/SOGo/LDAPSource.m b/SoObjects/SOGo/LDAPSource.m
--- a/SoObjects/SOGo/LDAPSource.m	2017-03-24 14:46:32.000000000 +0100
+++ b/SoObjects/SOGo/LDAPSource.m	2017-03-31 03:34:18.352544007 +0200
@@ -106,6 +106,8 @@
       listRequiresDot = YES;
 
       searchAttributes = nil;
+      LDAPLookupFields = [NSArray arrayWithObject: @"*"];
+      [LDAPLookupFields retain];
       passwordPolicy = NO;
       updateSambaNTLMPasswords = NO;
 
@@ -150,6 +152,7 @@
   [modulesConstraints release];
   [_scope release];
   [searchAttributes release];
+  [LDAPLookupFields release];
   [domain release];
   [kindField release];
   [multipleBookingsField release];
@@ -191,6 +194,7 @@
        IMAPLoginField: [udSource objectForKey: @"IMAPLoginFieldName"]
        SieveHostField: [udSource objectForKey: @"SieveHostFieldName"]
            bindFields: [udSource objectForKey: @"bindFields"]
+     LDAPLookupFields: [udSource objectForKey: @"LDAPLookupFields"]
             kindField: [udSource objectForKey: @"KindFieldName"]
             andMultipleBookingsField: [udSource objectForKey: @"MultipleBookingsFieldName"]];
 
@@ -313,6 +317,7 @@
     IMAPLoginField: (NSString *) newIMAPLoginField
     SieveHostField: (NSString *) newSieveHostField
         bindFields: (id) newBindFields
+  LDAPLookupFields: (NSArray *) newLDAPLookupFields
          kindField: (NSString *) newKindField
   andMultipleBookingsField: (NSString *) newMultipleBookingsField
 {
@@ -358,6 +363,8 @@
           ASSIGN(bindFields, [newBindFields componentsSeparatedByString: @","]);
         }
     }
+  if (newLDAPLookupFields)
+    ASSIGN(LDAPLookupFields, newLDAPLookupFields);
   if (newKindField)
     ASSIGN(kindField, [newKindField lowercaseString]);
   if (newMultipleBookingsField)
@@ -1203,7 +1210,6 @@
   NSEnumerator *entries;
   NSMutableArray *contacts;
   EOQualifier *qualifier;
-  NSArray *attributes;
 
   contacts = [NSMutableArray array];
 
@@ -1211,20 +1217,19 @@
     {
       ldapConnection = [self _ldapConnection];
       qualifier = [self _qualifierForFilter: match];
-      attributes = [NSArray arrayWithObject: @"*"];
 
       if ([_scope caseInsensitiveCompare: @"BASE"] == NSOrderedSame)
         entries = [ldapConnection baseSearchAtBaseDN: baseDN
                                            qualifier: qualifier
-                                          attributes: attributes];
+                                          attributes: LDAPLookupFields];
       else if ([_scope caseInsensitiveCompare: @"ONE"] == NSOrderedSame)
         entries = [ldapConnection flatSearchAtBaseDN: baseDN
                                            qualifier: qualifier
-                                          attributes: attributes];
+                                          attributes: LDAPLookupFields];
       else /* we do it like before */ 
         entries = [ldapConnection deepSearchAtBaseDN: baseDN
                                            qualifier: qualifier
-                                          attributes: attributes];
+                                          attributes: LDAPLookupFields];
       while ((currentEntry = [entries nextObject]))
         [contacts addObject:
                     [self _convertLDAPEntryToContact: currentEntry]];
@@ -1237,23 +1242,21 @@
 {
   NGLdapConnection *ldapConnection;
   NSEnumerator *entries;
-  NSArray *attributes;
 
   ldapConnection = [self _ldapConnection];
-  attributes = [NSArray arrayWithObject: @"*"];
 
   if ([_scope caseInsensitiveCompare: @"BASE"] == NSOrderedSame)
     entries = [ldapConnection baseSearchAtBaseDN: baseDN
                                        qualifier: qualifier
-                                      attributes: attributes];
+                                      attributes: LDAPLookupFields];
   else if ([_scope caseInsensitiveCompare: @"ONE"] == NSOrderedSame)
     entries = [ldapConnection flatSearchAtBaseDN: baseDN
                                        qualifier: qualifier
-                                      attributes: attributes];
+                                      attributes: LDAPLookupFields];
   else
     entries = [ldapConnection deepSearchAtBaseDN: baseDN
                                        qualifier: qualifier
-                                      attributes: attributes];
+                                      attributes: LDAPLookupFields];
 
   return [entries nextObject];
 }
@@ -1742,6 +1745,7 @@
 		  IMAPLoginField: nil
 		  SieveHostField: nil
 		 bindFields: nil
+	LDAPLookupFields: nil
 		  kindField: nil
 		  andMultipleBookingsField: nil];
               [ab setListRequiresDot: NO];
Only in b/SoObjects/SOGo: LDAPSource.m.orig
Only in b/SoObjects/SOGo: LDAPSource.m.rej
